lib/pbench/server/auth/__init__.py

```diff
@@ -115,18 +115,30 @@ def set_well_known_endpoints(self):
             )
             raise
 
-    def get_oidc_public_key(self, token: str):
+    def _get_oidc_public_key(self) -> str:
+        """Returns the public key of the current OIDC client.
+
+        Returns:
+            public key string
+
+        Raises:
+            KeyError: if public key not found in response payload
+            requests.exceptions: Requests session Exceptions
```

Comment:
This is awkward to read since it's so generic, and `requests.exceptions` refers to a Python *module* that defines exceptions, not to any specific exception.

However it occurs to me that, as Webb is fond of pointing out, **this** method doesn't raise any of these exceptions anyway, and shouldn't be responsible for documenting exceptions that might be raised in methods it calls. This all comes from `_get` and should be documented (a bit more clearly) there. (And whether it currently does or not arguably is not an issue for this PR. 😆 )

Reply:
Agreed.  Although, the text here could go so far as to point out that the caller could receive any exceptions raised by `_get()` and so (properly) pass the buck that way.